tick/ast/node.go

```diff
@@ -981,6 +1024,101 @@ func newProgram(p position) *ProgramNode {
 	}
 }
 
+func NewProgramNodeFromTickscript(tickscript string) (*ProgramNode, error) {
+	p, err := Parse(tickscript)
+
+	if err != nil {
+		return nil, fmt.Errorf("invalid TICKscript: %v", err)
+	}
+
+	pn, ok := p.(*ProgramNode)
+	// This should never happen
+	if !ok {
+		return nil, errors.New("invalid TICKscript")
+	}
+
+	return pn, nil
+}
+
+// TODO: Not sure how I feel about client.TaskType importing from client
+func (n *ProgramNode) DBRPs() []client.DBRP {
```

Comment:
In general importing the client types in the server code makes sense as it ensures that the server and client are using the same types. But here we are not server code but rather an independent package that handles parsing scripts.

I recommend that you use the AST nodes as the public API from the `tick` package and move these methods into the server code where needed.  So it would look something like `dbrpsFromProgram(*ast.ProgramNode) []client.DBRP`. Does that help?